packages/backend/src/clients/cube/CubeClient.ts

```diff
@@ -133,20 +138,48 @@ export default class CubeClient implements SemanticLayerClient {
         query: SemanticLayerQuery,
         callback: (results: SemanticLayerResultRow[]) => void,
     ): Promise<number> {
```

Comment:
I don't get the signature here? Shouldn't it return a readable stream?

Reply:
This function receives a callback that will handle the results, the write stream comes from `streamResultsToLocalFile` in `packages/backend/src/models/DownloadFileModel.ts`
The logic where we pass it down is in `packages/backend/src/services/SemanticLayerService/SemanticLayerService.ts` `streamQueryIntoFile` function